### DISCUSSION TITLE
Firebase Subcollections

### DISCUSSION THREAD

**Discussion by RSelwa** [maxnowack/signaldb#1025] [Ideas]
It would be cool to work with firebase subcollection to write and update data 😍

**Comment by maxnowack** [comment]
I'm not that familiar with firebase. What exactly are you trying to do with it and how would it be used with SignalDB?

**Reply by RSelwa** [reply]
We're working with firebase and we're trying to implement an ORM, but firebase works with sub-collection concepts. For exemple, our users have a sub coll of their preferences so basically the request should be something like '/users/id_user/preferences/id_of_preferences'. But in signalDb (as far as I know) you need to declare the collection path at initialization with the `createFirebasePersistenceAdapter`. But here we need to know the userId first to declare the good path. 
Currently, we'me made a little trick: we declared some collections knowed as sub-coll and if this is the case, we pass a ref in the object (wich gonna be `/user/<XYZ>/preferences/`) and we overide the collection path declared before with it.

Let me know what do you think about it 🤗

**Reply by maxnowack** [reply]
I've updated the examples last week to use the new `SyncManager` (see https://github.com/maxnowack/signaldb/blob/main/examples/firebase/src/system/syncManager.ts).
Please also take a look at the [implementation documentation](https://signaldb.js.org/sync/implementation/). There is an example how to specify an api path when using a REST API. For your use case it could be used similar

**Reply by RSelwa** [reply]
Yes, it work perfectly fine for collections, but for sub-collections, it's not possible without doing tricks. For Firestore, it's not like an REST API. 
It's collection name, with sub-collection, we can't declare it in the sync manager. Here's an example:
 ``` js
 registerRemoteChange({ name }, onChange) {
    onSnapshot(collection(db, name), () => {
      void onChange()
    })

    return Promise.resolve()
  },
```

We don't know the id of the document wich our sub-collection is from

**Reply by Cmacu** [reply]
You can pass extra parameters to `syncManager.addCollection` and you should receive them in the first argument of `registerRemoteChange`. See here: [Adding collections](https://signaldb.js.org/sync/implementation/#adding-collections)
